.github/workflows/ios_deploy_to_staging.yml

```diff
@@ -0,0 +1,63 @@
+name: ios-deploy-to-staging
+on:
+  # Trigger the workflow on push action
+  push:
+    branches:
+      - feature/48-as-a-developer-i-can-have-a-cd-process-to-build-and-deploy-the-ios-app-to-firebase-app-distribution
+
+jobs:
+  build_and_upload_staging_app_to_firebase:
+    name: Build And Upload iOS Staging Application To Firebase
+    runs-on: macOS-latest
+    defaults:
+      run:
+        working-directory: .template
+    env:
+      TEAM_ID: ${{ secrets.TEAM_ID }}
+      FASTLANE_USER: ${{ secrets.FASTLANE_USER }}
+      FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
+      FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD: ${{ secrets.FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD }}
+      FASTLANE_SESSION: ${{ secrets.FASTLANE_SESSION }}
+      MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
+      KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Install SSH key
+      uses: webfactory/ssh-agent@v0.4.1
+      with:
+        ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+    - uses: subosito/flutter-action@v1
+      with:
+        channel: 'stable'
+        flutter-version: '3.0.4'
+
+    - name: Get flutter dependencies
+      run: flutter pub get
+
+    - name: Run code generator
+      run: flutter packages pub run build_runner build --delete-conflicting-outputs
+
+    - name: Bundle install
+      run: cd ./ios && bundle install
+
+    - name: Pod install
+      run: cd ./ios && pod install
+
+    - name: Match Adhoc
+      run: cd ./ios && bundle exec fastlane sync_adhoc_staging_signing
+
+    - name: Deploy to Firebase
+      run: |
+        cd ./ios && bundle exec fastlane build_and_upload_staging_app
+      env:
+        FIREBASE_CLI_TOKEN: ${{ secrets.FIREBASE_CLI_TOKEN }}
```

Comment:
@markgravity Should we use [`flutter build ipa`](https://docs.flutter.dev/deployment/ios#create-an-app-bundle) command instead? Please check out my approach from the Varun project.

https://github.com/nimblehq/varun-flutter/blob/c2229cba27d757487299a9254327dd8f506d7003/bitrise.yml#L94-L113

Reply:
@luongvo We better to write them in the Fastlane, so that we can easier to use on the other, such as CodeMagic, Bitrise...
What do you think? 🤔

Reply:
@markgravity Agree 👍